docs/01 - Contributing/A - Development Environment.md

```diff
@@ -0,0 +1,203 @@
+# How to set up the development environment
+
+<!-- TOC -->
+* [How to set up the development environment](#how-to-set-up-the-development-environment)
+  * [Install JDK 11](#install-jdk-11)
+  * [Checkout the Project](#checkout-the-project)
+  * [Set up `docker`](#set-up-docker)
+  * [Set up your hosts file](#set-up-your-hosts-file)
+  * [Build the Project with Gradle](#build-the-project-with-gradle)
+    * [Clean](#clean)
+    * [Build](#build)
+    * [Running Unit Tests](#running-unit-tests)
+  * [Set up the Git Hooks](#set-up-the-git-hooks)
+* [Set up the Development Environment with IntelliJ IDEA](#set-up-the-development-environment-with-intellij-idea)
+  * [Configuring Gradle on IntelliJ IDEA](#configuring-gradle-on-intellij-idea)
+  * [Run Configurations](#run-configurations)
+    * [Example: Debug the Platform Server](#example-debug-the-platform-server)
+    * [Example: Debug the Junit or the End-to-End Tests](#example-debug-the-junit-or-the-end-to-end-tests)
+  * [Running the Tests From Gradle in IntelliJ](#running-the-tests-from-gradle-in-intellij)
+<!-- TOC -->
+
+## Install JDK 11
+
+Before you start, please make sure you
+have [JDK-11](https://www.oracle.com/java/technologies/javase-jdk11-downloads.html) installed on your machine.
+
+To check if you have it installed, run:
+
+```shell
+java -version
+```
+
+## Checkout the Project
+
+To check out the project, run:
+
+```shell
+git clone git@github.com:stellar/java-stellar-anchor-sdk.git 
+```
+
+or
+
+```shell
+git clone https://github.com/stellar/java-stellar-anchor-sdk.git
+```
+
+## Set up `docker`
+
+Please make sure you have `docker` installed on your machine. If you don't, please follow the instructions on
+the [docker website](https://docs.docker.com/get-docker/).
+
+Docker version 23.0.0 or higher is recommended. To check your docker version, run:
+
+```shell
+docker --version
+```
+
+You should see something like:
+
+```shell
+Docker version 23.0.5, build bc4487a
+```
+
+## Set up your hosts file
+
+On macOS and Linux, you need to add the following entries to your `/etc/hosts` file. And on Windows, you need to add the
+following entries to your `C:\Windows\System32\Drivers\etc\hosts` file.
+
+```shell
+127.0.0.1 db
+127.0.0.1 zookeeper
+127.0.0.1 kafka
+127.0.0.1 sep24-reference-ui
+127.0.0.1 reference-server
+127.0.0.1 wallet-server
+127.0.0.1 platform
+127.0.0.1 host.docker.internal
+```
```

Comment:
This has not been my experience. I've gotten everything working without adding these hosts. Can you reproduce?

Reply:
It's for tests

Reply:
Yeah, this is for integration and end-2-end tests where the servers communicate with the docker containers.

Reply:
Added the purpose of the DNS entries to the doc. 
https://github.com/stellar/java-stellar-anchor-sdk/pull/1064/files#diff-700ef4c73b87f78486e1b6c614d99548eebc17d61df8cc6613cc1e2b668ab615R71